### PR TITLE
[nextercism] cmd: Various simplifications

### PR DESCRIPTION
**Pull request by ferhatelmas**
* `if err != nil { return err } return nil`, just returns

there were 2 more issues:
* defer is called without checking error
* error wasn't checked on walk

### CODE REVIEW

File: cmd/configure.go [hunk 1]
```diff
@@ -62,12 +62,7 @@ You can also override certain default settings to suit your preferences.
 			return err
 		}
 
-		err = apiCfg.Write()
-		if err != nil {
-			return err
-		}
-
-		return nil
+		return apiCfg.Write()
```
Comment (nywilken) [inline cmd/configure.go]: Good catch on theses 👍

File: cmd/submit_test.go [hunk 2]
```diff
@@ -76,11 +76,10 @@ func TestSubmit(t *testing.T) {
 		files := mf.File["files[]"]
 		for _, fileHeader := range files {
 			file, err := fileHeader.Open()
-			defer file.Close()
-
 			if err != nil {
 				t.Fatal(err)
 			}
+			defer file.Close()
 			body, err := ioutil.ReadAll(file)
```
Comment (nywilken) [inline cmd/submit_test.go]: I believe by calling defer after checking the error you run the risk of stopping execution without ever closing the file.

Comment (kytrinyx) [reply cmd/submit_test.go]: If I understand the source code for`fileHeader.Open()` correctly, it calls `os.OpenFile()` (https://golang.org/src/os/file_unix.go?s=4493:4559#L137) behind the scenes, which will either return a file _or_ an error.

That means that if we do have an error, the file will be `nil`.

So it is safe to call `defer file.Close()` after dealing with the error.

**Comment by kytrinyx** [comment]
> Disregard the comment on defer - it's an open so if it fails there is no file handle to call Close on.

D'oh I only saw this after commenting. Disregard my comment about disregarding... well, whatever :)

This looks great. Thanks @ferhatelmas

**Pull request merged**